src/util/tprint.cpp

```diff
@@ -48,17 +48,25 @@ u32 smek_snprint(char *out_buffer, u32 buf_size, const char *in_buffer) {
 template <>
 i32 sntprint<>(char *buffer, u32 buf_size, const char *fmt) {
     if (buf_size == 0) return 0;
+    u32 write = 0;
     u32 head = 0;
-    while (fmt[head] && head != buf_size) {
-        buffer[head] = fmt[head];
-        head++;
+    for (; fmt[head] && head != buf_size; head++) {
+        if (fmt[head] == '%') {
+            if (fmt[head + 1] == '{') { continue; }
+        } else if (fmt[head] == '{') {
+            WARN("Invalid format string, unexpected '{}'", fmt + head);
+            buffer[write] = '\0';
```

Comment:
Maybe we could print the invalid string or (even better) on which line/file it was found. At least we could do it with the log-macros and have default parameters otherwise (e.g., line 0 => no line number found so we print the string instead).

Reply:
We don't know the line we were called from, unless we dump the backtrace. And we don't know the entire string, only the part that is passed in here... There's a lot of incomplete information. Maybe do something smart on the return and risk getting multiple logs? There isn't really a good solution here.

Reply:
Hmmm yeah that's true... As long as we don't dynamically add arguments I guess we're good, since it should be easy to check what was added since the run.